src/circuit/mod.rs

```diff
@@ -1301,9 +1306,27 @@ mod test {
         (verification_key, proof)
     }
 
+    fn check_if_crs_exists_and_download(n: usize) {
+        let file_path = &format!("setup_2^{}.key", n);
```

Comment:
You have a problem here. You check if a file is in this path, but then you save it in a different one. To avoid this error, simply agree in a path, and remove the `let crx_file_destination = //` in line 1316.